composer/datasets/cifar.py

```diff
@@ -49,10 +68,84 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
                 memory_format=self.synthetic_memory_format,
             )
 
+        elif self.use_ffcv:
+            try:
+                import ffcv  # type: ignore
+                from ffcv.fields.decoders import IntDecoder, SimpleRGBImageDecoder  # type: ignore
+                from ffcv.pipeline.operation import Operation  # type: ignore
+            except ImportError:
+                raise ImportError(
+                    textwrap.dedent("""\
+                    Composer was installed without ffcv support.
+                    To use ffcv with Composer, run `pip install mosaicml[ffcv]`"""))
+
+            dataset_file = self.ffcv_dest_train if self.is_train else self.ffcv_dest_val
+            dataset_filepath = os.path.join(self.ffcv_dir, dataset_file)
+            # always create if ffcv_write_dataset is true
+            if self.ffcv_write_dataset and not os.path.exists(dataset_filepath):
```

Comment:
Would this have a race condition with multi-rank training?

Reply:
Yes. will fix this.

Reply:
Only rank 0 creates it.